providers/dns/servercow/servercow.go

```diff
@@ -0,0 +1,223 @@
+// Package servercow implements a DNS provider for solving the DNS-01 challenge using Servercow DNS.
+package servercow
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+	"github.com/go-acme/lego/v3/providers/dns/servercow/internal"
+)
+
+const defaultTTL = 120
+
+// Config is used to configure the creation of the DNSProvider.
+type Config struct {
+	Username string
+	Password string
+
+	PropagationTimeout time.Duration
+	PollingInterval    time.Duration
+	TTL                int
+	HTTPClient         *http.Client
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	return &Config{
+		TTL:                env.GetOrDefaultInt("SERVERCOW_TTL", defaultTTL),
+		PropagationTimeout: env.GetOrDefaultSecond("SERVERCOW_PROPAGATION_TIMEOUT", dns01.DefaultPropagationTimeout),
+		PollingInterval:    env.GetOrDefaultSecond("SERVERCOW_POLLING_INTERVAL", dns01.DefaultPollingInterval),
+		HTTPClient: &http.Client{
+			Timeout: env.GetOrDefaultSecond("SERVERCOW_HTTP_TIMEOUT", 30*time.Second),
+		},
+	}
+}
+
+// DNSProvider implements challenge.Provider for the Servercow API.
+type DNSProvider struct {
+	config *Config
+	client *internal.Client
+}
+
+// NewDNSProvider returns a DNSProvider instance.
+func NewDNSProvider() (*DNSProvider, error) {
+	values, err := env.Get("SERVERCOW_USERNAME", "SERVERCOW_PASSWORD")
+	if err != nil {
+		return nil, fmt.Errorf("servercow: %w", err)
+	}
+
+	config := NewDefaultConfig()
+	config.Username = values["SERVERCOW_USERNAME"]
+	config.Password = values["SERVERCOW_PASSWORD"]
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for Servercow.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config.Username == "" || config.Password == "" {
+		return nil, fmt.Errorf("servercow: some credentials information are missing")
```

Comment:
The error message is a bit vague, better "servercow: incomplete credentials, missing username/password"?